src/components/Selector/BaseSelector/index.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { KeyboardEventHandler, useState } from "react";
+import { Autocomplete } from "@material-ui/lab";
+import { TextField } from "@material-ui/core";
+import styles from "./styles.module.scss";
+import classNames from "classnames";
+
+export const BaseSelector = <Option, Value>(
+  {
+    label,
+    className,
+    getOptionValue,
+    initialValue,
+    errorMessage,
+    options,
+    ...autocompleteProps
+  }: IBaseSelectorProps<Option, Value>
+) => {
+  const [defaultValue] = useState(options.find(option => getOptionValue(option) === initialValue));
```

Comment:
Definis un estado pero no pensas cambiarlo? porque no definis una constante?

Reply:
quiero que se guarde el resultado de la primera vez que le cae initialValue

si usara una variable, estaría cambiando defaultValue en un componente uncontrolled, and that's no good

Reply:
![image](https://user-images.githubusercontent.com/20487262/79016616-ded9d880-7b45-11ea-9678-f54baf8f3110.png)